### PR TITLE
Change type of amount in Binance Timelock messages.

### LINKED ISSUE #1110: Add timelock support
**Is your feature request related to a problem? Please describe.**
It is very nice if we can run `timeLock`, `timeUnlock`, and `timeRelock` function via walletConnect.

https://github.com/binance-chain/javascript-sdk/blob/master/docs/api-docs/classes/tokenmanagement.md#timelock
https://docs.binance.org/timelock.html

**Describe the solution you'd like**
Maybe add `timelock_order`, `timerelock_order`, and `timeunlock_order` in here.

https://github.com/trustwallet/wallet-core/blob/61c55bfa7f49ec9c916d54aad6b4cd0c32051613/src/proto/Binance.proto#L173

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

In the recently-added Binance Timelock messages, the amount should not be a single amount, but an array of symbol-amount pairs.  Fixes #1110 .

## Testing instructions

Unit tests, `*Binance*`.

## Types of changes

* Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Review by vikmeup — APPROVED** [review]

File: docs/coins.md [hunk 1]
```diff
@@ -64,4 +64,5 @@ This list is generated from [./coins.json](../coins.json)
 | 19167   | Zelcash          | ZEL    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/zelcash/info/logo.png" width="32" />      | <https://zel.cash>            |
 | 5718350 | Wanchain         | WAN    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/wanchain/info/logo.png" width="32" />     | <https://wanchain.org>        |
 | 5741564 | Waves            | WAVES  | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/waves/info/logo.png" width="32" />        | <https://wavesplatform.com>   |
-| 20000714 | Smart Chain      | BNB    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/bsc/info/logo.png" width="32" />          | <https://www.binance.org/en/smartChain> |
+| 10000714 | Smart Chain Legacy | BNB    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/bsc/info/logo.png" width="32" />          | <https://www.binance.org/en/smartChain> |
```
Comment (hewigovens) [inline docs/coins.md:67]: @catenocrypt can you revert this file? @vikmeup maybe time to remove smart chain legacy from wc

File: src/proto/Binance.proto [hunk 2]
```diff
@@ -164,15 +165,17 @@ message SideChainUndelegate {
 message TimeLockOrder {
     bytes from_address = 1; // owner address
     string description = 2;
-    int64 amount = 3;
+    // Array of symbol/amount pairs.  Naming is a bit misleading, but we follow SDK https://github.com/binance-chain/javascript-sdk/blob/master/docs/api-docs/classes/tokenmanagement.md#timelock
```
Comment (hewigovens) [inline src/proto/Binance.proto:168]:
```suggestion
    // Array of symbol/amount pairs. see SDK https://github.com/binance-chain/javascript-sdk/blob/master/docs/api-docs/classes/tokenmanagement.md#timelock
```

File: src/proto/Binance.proto [hunk 3]
```diff
@@ -164,15 +165,17 @@ message SideChainUndelegate {
...
+    repeated SendOrder.Token amount = 3;
     int64 lock_time = 4;
 }
 
 message TimeRelockOrder {
     bytes from_address = 1; // owner address
     int64 id = 2; // order ID
     string description = 3;
-    int64 amount = 4;
+    // Array of symbol/amount pairs.  Naming is a bit misleading, but we follow SDK.
```
Comment (hewigovens) [inline src/proto/Binance.proto:177]:
```suggestion
    // Array of symbol/amount pairs.
```

**Review by hewigovens — APPROVED** [review]

File: docs/coins.md (hunk 1 above)
Comment (catenocrypt) [reply docs/coins.md]: I did revert here, but please note that this is a generated file and shows up in several PRs (e.g. in IOV PR).  If not updated, it is not in sync with coins.json.  Also, info is outdated, as refers to bsc folder for non-legacy 20000714 chain.

Comment (vikmeup) [reply docs/coins.md]: let's remove little later, 1-2 months from now?

File: src/proto/Binance.proto (hunk 2 above)
Comment (catenocrypt) [reply src/proto/Binance.proto]: ok

**Pull request merged**